api/api.go

```diff
@@ -437,6 +441,62 @@ type CloudBackupSchedEnumerateResponse struct {
 	Schedules map[string]CloudBackupScheduleInfo
 }
 
+type Operation string
+
+const (
+	MigrateCluster     = Operation("MigrateCluster")
+	MigrateVolume      = Operation("MigrateVolume")
+	MigrateVolumeGroup = Operation("MigrateVolumeGroup")
+)
+
+type MigrateStatus string
+
+const (
+	MigrateStatusQueued       = MigrateStatus("Queued")
+	MigrateStatusInitializing = MigrateStatus("Initializing")
+	MigrateStatusInProgress   = MigrateStatus("InProgress")
+	MigrateStatusFailed       = MigrateStatus("Failed")
+	MigrateStatusComplete     = MigrateStatus("Complete")
+)
+
+type MigrateStage string
+
+const (
+	MigrateStageBackup       = MigrateStage("Backup")
+	MigrateStageRestore      = MigrateStage("Restore")
+	MigrateStageVolumeUpdate = MigrateStage("VolumeUpdate")
+	MigrateStageDone         = MigrateStage("Done")
+)
+
+type CloudMigrateInfo struct {
+	ClusterId       string
+	LocalVolumeId   string
+	LocalVolumeName string
+	RemoteVolumeId  string
+	CloudSnapId     string
+	CurrentStage    MigrateStage
+	Status          MigrateStatus
+	LastUpdate      time.Time
+	LastSuccess     time.Time
+}
+
+type CloudMigrateStartRequest struct {
```

Comment:
Move structs as messages in the api.proto file.

Reply:
They depend on the typed consts above, so can't move them there

Reply:
They need to move to have them used by the SDK in the future.